### PR TITLE
add new feature

### PR DESCRIPTION
**Pull request by AnatoliiLykhohodenko**
to check you need to run this code locally (npm start) and open a window [http://localhost:5000/products?limit=1&offset=20](http://localhost:5000/products?limit=1&offset=20) (also try with different queries)

### CODE REVIEW

**Review by kochetov-yevhen-may29 — APPROVED** [review]
Gj)

File: package.json [hunk 1]
```diff
@@ -4,10 +4,10 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "start": "nodemon index.ts",
+    "start": "npx ts-node index.ts",
```
Comment (ilona-borkovska) [inline package.json:7]: do we need npx here?

File: src/server.ts [hunk 2]
```diff
@@ -13,8 +14,6 @@ app.use(cors({
 }));
 app.use(express.json());
 
-app.use('/', (req, res) => {
-  res.send('Product catalog');
-});
+app.use(productsRouter);
```
Comment (ilona-borkovska) [inline src/server.ts:17]:
```suggestion
app.use("/products", productsRouter);
```

File: src/routes/products.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import * as ProductController from '../controllers/products';
+import express from 'express';
+
+export const router = express.Router();
+
+router.get('/products', ProductController.getAll)
+router.get('/products/:id', ProductController.getById)
```
Comment (ilona-borkovska) [inline src/routes/products.ts:7]:
```suggestion
router.get('/', ProductController.getAll)
router.get('/:productId', ProductController.getById)
```

File: src/controllers/products.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { Model } from 'sequelize';
+import * as ProductsService from '../services/product';
+import { Request, Response } from 'express';
+
+export async function getAll(
+  req: Request,
+  res: Response,
+) {
+  const limit = req.query.limit ? +req.query.limit : 10;
+  const offset = req.query.offset ? +req.query.offset : 0;
```
Comment (ilona-borkovska) [inline src/controllers/products.ts:10]:
```suggestion
  const limit = !isNaN(+req.query.limit) ? +req.query.limit : 10;
  const offset = !isNaN(+req.query.offset) ? +req.query.offset : 0;
```

File: package.json [hunk 5]
```diff
@@ -4,10 +4,10 @@
...
     "build": "tsc",
     "generate-seeder-file": " npx sequelize-cli seed:generate --name demo-product",
-    "db-seed": "npx sequelize-cli db:seed:all"
+    "db-seed": "npx sequelize-cli db:seed:all",
```
Comment (ilona-borkovska) [inline package.json:10]: the same

File: src/controllers/products.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  if (res) {
+    res.json(result);
+  }
+}
+
+export async function getById(
+  req: Request,
+  res: Response,
+) {
+  const result = await ProductsService.getById(+req.params.id);
```
Comment (ilona-borkovska) [inline src/controllers/products.ts:23]:
```suggestion
  const result = await ProductsService.getById(+req.params.productId);
```

File: src/services/product.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { Product } from "../utils/db_product_table";
```
Comment (ilona-borkovska) [inline src/services/product.ts:1]: products.ts - change file name

**Review by ilona-borkovska — APPROVED** [review]
Nice))

**Pull request merged**